KAFKA-14059 Replace PowerMock with Mockito in WorkerSourceTaskTest

hgeraldino opened the pull request:
Follows more or less the same principles as https://github.com/apache/kafka/pull/13191

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

hgeraldino committed:
KAFKA-14059 Replace EasyMock and PowerMock with Mockito in WorkerSourceTaskTest

hgeraldino committed:
KAFKA-14059 Replace EasyMock and PowerMock with Mockito in WorkerSourceTaskTest

hgeraldino committed:
Few tweaks to make JDK8 builds happy

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,166 +746,117 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
+        assertEquals(TOPIC, topicCapture.getValue());
     }
 
     private boolean awaitLatch(CountDownLatch latch) {
```
Out of scope, so implement (or don't) at your discretion, but this method (and invocations of it) can be replaced with [ConcurrencyUtils::awaitLatch](https://github.com/apache/kafka/blob/1d8f79964e9578f3769fd13c2535988bfc803696/connect/runtime/src/test/java/org/apache/kafka/connect/test/util/ConcurrencyUtils.java#L42-L50).

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -315,33 +305,28 @@ public void testPause() throws Exception {
 
-        PowerMock.verifyAll();
+        verifyCleanStartup();
+        verifyTaskGetTopic(10);
```
Should this use `count.get()` instead of `10`, since it's possible that the task may have been polled extra times between when line 293 (`assertTrue(awaitLatch(pollLatch));`) completed and when the task finished pausing?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -315,33 +305,28 @@ public void testPause() throws Exception {
         // In this test, we don't flush, so nothing goes any further than the offset writer
 
         expectTopicCreation(TOPIC);
+        expectOffsetFlush(true, true);
```
Nit: this can be simplified, right?
```suggestion
        expectOffsetFlush(true);
```

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -315,33 +305,28 @@ public void testPause() throws Exception {
-        expectCleanStartup();
-
         final CountDownLatch pollLatch = expectPolls(10);
         // In this test, we don't flush, so nothing goes any further than the offset writer
```
Is this still accurate? Looks like we do end up performing an offset flush in this test.

hgeraldino replied:
you're right, removed comment

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -1003,31 +868,35 @@ private boolean awaitLatch(CountDownLatch latch) {
+    private void expectOffsetFlush(boolean succeed, boolean offsetFlush, Boolean... remainingVals) throws Exception {
+        when(offsetWriter.beginFlush(anyLong(), any(TimeUnit.class))).thenReturn(offsetFlush, remainingVals);
+
+        if (offsetFlush) {
```
Isn't this technically incorrect? Shouldn't we set up expectations for `offsetWriter::doFlush` if any of `remainingVals` is `true` as well?

hgeraldino replied:
This was definitely wrong. I re-implemented this logic splitting the `beginFlush` and `doFlush` calls into separate util methods.

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -391,37 +369,26 @@ public void testFailureInPoll() throws Exception {
+        verifyCleanStartup();
+        verifyOffsetFlush(true);
+        verify(statusListener).onFailure(taskId, exception);
+        verify(sourceTask).stop();
```
Looks like we might be missing coverage for this call (originally verified in `expectEmptyOffsetFlush`):
```suggestion
        verify(sourceTask).commit();
        verify(sourceTask).stop();
```

C0urante replied:
Believe this still needs to be addressed?

C0urante replied:
Ah, never mind. This is covered by changes to `verifyOffsetFlush`.

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -433,35 +400,36 @@ public void testFailureInPollAfterCancel() throws Exception {
+        verify(sourceTask).stop();
+        verify(admin).close(any(Duration.class));
+        verify(transformationChain).close();
+        verify(offsetReader, times(2)).close();
```
Since `OffsetStorageReaderImpl::close` is idempotent, it doesn't hurt that we invoke it twice, but it's definitely not necessary. We just have to make sure it's invoked at least once:
```suggestion
        verify(offsetReader, atLeast(1)).close();
```

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -474,129 +442,107 @@ public void testFailureInPollAfterStop() throws Exception {
         assertTrue(awaitLatch(pollLatch));
         assertTrue(workerTask.commitOffsets());
+        verify(offsetWriter).beginFlush(anyLong(), any(TimeUnit.class));
+        reset(offsetWriter);
```
I try to avoid `reset` when I can. Can we remove this line and change the next assertion for `offsetWriter::beginFlush` to use `times(2)`?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -474,129 +442,107 @@ public void testFailureInPollAfterStop() throws Exception {
         final CountDownLatch pollLatch = expectPolls(1);
-        expectOffsetFlush(true);
-
-        offsetWriter.offset(PARTITION, OFFSET);
```
Are we losing coverage for this invocation in this test?

hgeraldino replied:
Not anymore

C0urante reviewed (commented):
Thanks @hgeraldino! I've made it about halfway through in this pass; will try to revisit this week or the next and look at the rest.

Also, similar to https://github.com/apache/kafka/pull/13191#discussion_r1097852548, can we add `verifyNoMoreInteractions(statusListener);` to the test's `tearDown` method?

hgeraldino commented:
> Thanks @hgeraldino! I've made it about halfway through in this pass; will try to revisit this week or the next and look at the rest.
> 
> Also, similar to [#13191 (comment)](https://github.com/apache/kafka/pull/13191#discussion_r1097852548), can we add `verifyNoMoreInteractions(statusListener);` to the test's `tearDown` method?

Good catch, will add it.

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino committed:
PR feedback

hgeraldino commented:
Thanks for your feedback @C0urante. 

I finally had time this past week and tweaked the test to address your comments and fix coverage.

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -706,95 +673,77 @@ public void testSourceTaskIgnoresProducerException() throws Exception {
+        doAnswer((Answer<Object>) invocation -> {
             startupLatch.countDown();
-            assertTrue(awaitLatch(finishStartupLatch));
+            ConcurrencyUtils.awaitLatch(finishStartupLatch, POLL_TIMEOUT_MSG);
```
Shouldn't this get a different message?

hgeraldino replied:
Yes, that was lazy of me not to rewrite these messages 🤦

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -315,157 +322,130 @@ public void testPause() throws Exception {
+        when(sourceTask.poll()).thenAnswer(invocation -> {
             pollLatch.countDown();
-            assertTrue(awaitLatch(workerCancelLatch));
+            ConcurrencyUtils.awaitLatch(workerCancelLatch, POLL_TIMEOUT_MSG);
```
Shouldn't this get a different message?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -315,157 +322,130 @@ public void testPause() throws Exception {
+        when(sourceTask.poll()).thenAnswer(invocation -> {
             pollLatch.countDown();
-            assertTrue(awaitLatch(workerStopLatch));
+            ConcurrencyUtils.awaitLatch(workerStopLatch, POLL_TIMEOUT_MSG);
```
Shouldn't this get a different message?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -706,95 +673,77 @@ public void testSourceTaskIgnoresProducerException() throws Exception {
         // exiting the work thread immediately, and the stop() method will be invoked in the background thread since it
         // cannot be invoked immediately in the thread trying to stop the task.
-        assertTrue(awaitLatch(startupLatch));
+        ConcurrencyUtils.awaitLatch(startupLatch, POLL_TIMEOUT_MSG);
```
Shouldn't this get a different message?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -474,129 +454,120 @@ public void testFailureInPollAfterStop() throws Exception {
+        verifyCleanStartup();
+        verifyTopicCreation(TOPIC);
+        verify(offsetWriter, times(2)).beginFlush(anyLong(), any(TimeUnit.class));
+        verify(offsetWriter).offset(PARTITION, OFFSET);
```
Running this test locally, there's about a 25% rate of flaky failure here because `offsetWriter::offset` has been invoked too many times. Possible root cause:

On `trunk` we use `atLeastOnce` when [setting up expectations](https://github.com/apache/kafka/blob/02fb4b882b86bc835cf4b0bbdaee099d6056799b/connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java#L527-L528) for this method call. This is because two attempts are made to commit offsets by the task, both of which may cause calls to this method: the direct call to `workerTask::commitOffsets`, and the end-of-life offset commit made by the task during shutdown.

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -474,129 +454,120 @@ public void testFailureInPollAfterStop() throws Exception {
-        PowerMock.verifyAll();
+        verifyCleanStartup();
+        verify(sourceTask).stop();
+        verify(offsetWriter).offset(PARTITION, OFFSET);
```
Same flaky failure issues due to too many invocations

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +776,165 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
-        EasyMock.expect(offsetWriter.beginFlush(EasyMock.anyLong(), EasyMock.anyObject())).andReturn(true);
-        Future<Void> flushFuture = PowerMock.createMock(Future.class);
-        EasyMock.expect(offsetWriter.doFlush(EasyMock.anyObject(Callback.class))).andReturn(flushFuture);
+    private void expectOffsetFlush(List<Boolean> succeedList) throws Exception {
```
Can we use variadic arguments (i.e., `Boolean... successes` instead of `List<Boolean> succeedList`) here, to make things simpler for callers?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -474,129 +454,120 @@ public void testFailureInPollAfterStop() throws Exception {
-        statusListener.onShutdown(taskId);
-        EasyMock.expectLastCall();
-
-        expectClose();
-
-        PowerMock.replayAll();
+            @Override
+            public Boolean get() {
+                return succeed.next();
+            }
+        });
```
This can be simplified:

```java
        expectBeginFlush(Arrays.asList(true, false).iterator()::next);
```

Though it may be cleaner to introduce a new `expectBeginFlush` variant that accepts a `Boolean... successes` argument and handles all the iterator wrangling.

hgeraldino replied:
neat

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +776,165 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
+    }
+
+    private void expectBeginFlush(Supplier<Boolean> resultSupplier) throws Exception {
+        when(offsetWriter.beginFlush(anyLong(), any(TimeUnit.class))).thenReturn(resultSupplier.get());
```
This will invoke `resultSupplier::get` immediately and use that result for every invocation of `offsetWriter::beginFlush`.

If we want to use `resultSupplier` every time `offsetWriter::beginFlush` is invoked, we can do this:

```suggestion
        when(offsetWriter.beginFlush(anyLong(), any(TimeUnit.class))).thenAnswer(ignored -> resultSupplier.get());
```

Interestingly, all tests pass with and without this change.

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +776,165 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
 
-    private Capture<ProducerRecord<byte[], byte[]>> expectSendRecordAnyTimes() throws InterruptedException {
-        return expectSendRecordTaskCommitRecordSucceed(true);
+    //
```
Empty comment?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +776,165 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
-    private void expectTaskCommitRecordWithOffset(boolean anyTimes, boolean succeed) throws InterruptedException {
-        sourceTask.commitRecord(EasyMock.anyObject(SourceRecord.class), EasyMock.anyObject(RecordMetadata.class));
-        IExpectationSetters<Void> expect = EasyMock.expectLastCall();
+    //
```
Empty comment?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +776,165 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
             // 2. As a result of a successful producer send callback, we'll notify the source task of the record commit
-            expectTaskCommitRecordWithOffset(anyTimes, commitSuccess);
-            expectTaskGetTopic(anyTimes);
+            expectTaskCommitRecordWithOffset(commitSuccess);
```
We never call `expectSendRecord` with `sendSuccess` set to `true` and `commitSuccess` set to false. Should we eliminate the `commitSuccess` parameter and change this line to unconditionally invoke `expectTaskCommitRecordWithOffset(true)`?

Edit: actually, after examining `expectTaskCommitRecordWithOffset` and its call sites, I wonder if that method is effectively a no-op right now anyways.

hgeraldino replied:
It was.

Removed

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +776,165 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
-        sourceTask.commitRecord(EasyMock.anyObject(SourceRecord.class), EasyMock.anyObject(RecordMetadata.class));
-        IExpectationSetters<Void> expect = EasyMock.expectLastCall();
+    //
+    private void expectTaskCommitRecordWithOffset(boolean succeed) throws InterruptedException {
```
It looks like this method is only ever invoked with `true` as its argument. Do we even need it anymore?

hgeraldino replied:
Looks like we don't. Removed

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -706,95 +673,77 @@ public void testSourceTaskIgnoresProducerException() throws Exception {
+        expectPreliminaryCalls();
 
-        PowerMock.replayAll();
+        when(producer.send(any(ProducerRecord.class), any(Callback.class))).thenAnswer(
+                producerSendAnswer(true)
+        ).thenAnswer(producerSendAnswer(false));
```
Nit: A small whitespace change can help readability here.

```suggestion
        when(producer.send(any(ProducerRecord.class), any(Callback.class)))
                .thenAnswer(producerSendAnswer(true))
                .thenAnswer(producerSendAnswer(false));
```

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -706,95 +673,77 @@ public void testSourceTaskIgnoresProducerException() throws Exception {
+        verify(offsetWriter).offset(PARTITION, offset2);
+        verify(sourceTask).commitRecord(any(SourceRecord.class), isNull());
+
+        List<Field> fields = ReflectionUtils.findFields(WorkerSourceTask.class, field ->
+                field.getName().equals("submittedRecords"), ReflectionUtils.HierarchyTraversalMode.TOP_DOWN);
+        SubmittedRecords submittedRecords = (SubmittedRecords) ReflectionUtils.tryToReadFieldValue(fields.iterator().next(), workerTask)
+                .getOrThrow(RuntimeException::new);
```
Do we need to use reflection for this, or can we upgrade the visibility of the `submittedRecords` field to package-private?

hgeraldino replied:
Makes sense... changed

C0urante reviewed (commented):
Thanks @hgeraldino! Apologies for the wait, releases, Kafka Summit London, and a few other things have made it difficult to find for review over the last few weeks.

I've finished a complete pass over the PR; once everything is addressed this should be good to merge.

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino commented:
Thanks @C0urante for your thoroughly review! I think I've addressed all your comments. 

I've also picked up the work to have the `WorkerSinkTaskTest` migrated next (this one is a bit trickier), hope to have a PR  soon

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -706,95 +662,72 @@ public void testSourceTaskIgnoresProducerException() throws Exception {
+        doAnswer((Answer<Object>) invocation -> {
             startupLatch.countDown();
-            assertTrue(awaitLatch(finishStartupLatch));
+            ConcurrencyUtils.awaitLatch(finishStartupLatch, "Timeout waiting for task to stop");
```
Are we really waiting for the task to stop here?

hgeraldino replied:
Hmm... not really. I mean, the countDownLatch is updated after calling `workerTask.stop()` on L710, but calling `stop()` only sets an internal flag on the `WorkerTask` and doesn't really block, so technically no.

I don't understand the purpose of this synchronization, so decided to remove `finishStartupLatch` altogether

C0urante replied:
The purpose of `finishStartupLatch` is to ensure that we invoke `WorkerTask::stop` while the `WorkerTask` (on its separate thread) is in the middle of invoking `SourceTask::start`.

We should add that logic back. I was only commenting on the error message; the rest of the test case looked correct.

hgeraldino replied:
Ok, restored the latch and changed the verbiage to "finish" instead of "stop"

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -474,129 +451,113 @@ public void testFailureInPollAfterStop() throws Exception {
+        verifyCleanStartup();
+        verifyTopicCreation(TOPIC);
+        verify(offsetWriter, times(2)).beginFlush(anyLong(), any(TimeUnit.class));
+        verify(offsetWriter).offset(PARTITION, OFFSET);
```
We still have to update this to `atLeastOnce()`, right?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +760,155 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
+    private void expectSendRecord(
+            String topic,
+            boolean sendSuccess,
+            boolean commitSuccess,
```
Can we remove this parameter now?

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -832,202 +760,155 @@ private CountDownLatch expectPolls(int minimum, final AtomicInteger count) throw
+            String topic,
+            boolean sendSuccess,
+            boolean commitSuccess,
+            boolean isMockedConverters,
```
This is always `true`; can we remove it and unconditionally call `expectConvertHeadersAndKeyValue(topic, headers);` in the method body?

C0urante reviewed (commented):
Thanks @hgeraldino, really close!

hgeraldino committed:
PR feedback

hgeraldino committed:
PR feedback

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -315,157 +319,130 @@ public void testPause() throws Exception {
+        when(sourceTask.poll()).thenAnswer(invocation -> {
             pollLatch.countDown();
-            assertTrue(awaitLatch(workerCancelLatch));
+            ConcurrencyUtils.awaitLatch(workerCancelLatch, "Timeout waiting for task cancellation.");
```
```suggestion
            ConcurrencyUtils.awaitLatch(workerCancelLatch, "Timeout waiting for main test thread to cancel task.");
```

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -315,157 +319,130 @@ public void testPause() throws Exception {
+        when(sourceTask.poll()).thenAnswer(invocation -> {
             pollLatch.countDown();
-            assertTrue(awaitLatch(workerStopLatch));
+            ConcurrencyUtils.awaitLatch(workerStopLatch, "Timeout waiting for task to stop");
```
```suggestion
            ConcurrencyUtils.awaitLatch(workerStopLatch, "Timeout waiting for main test thread to stop task");
```

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -739,62 +693,42 @@ public void testSlowTaskStart() throws Exception {
+        doAnswer((Answer<Object>) invocation -> {
             startupLatch.countDown();
-            assertTrue(awaitLatch(finishStartupLatch));
+            ConcurrencyUtils.awaitLatch(finishStartupLatch, "Timeout waiting for task to finish");
```
```suggestion
            ConcurrencyUtils.awaitLatch(finishStartupLatch, "Timeout waiting for main test thread to allow task startup to complete");
```

C0urante commented on connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:
```diff
@@ -739,62 +693,42 @@ public void testSlowTaskStart() throws Exception {
         // exiting the work thread immediately, and the stop() method will be invoked in the background thread since it
         // cannot be invoked immediately in the thread trying to stop the task.
-        assertTrue(awaitLatch(startupLatch));
+        ConcurrencyUtils.awaitLatch(startupLatch, "Timeout waiting for task to start");
```
```suggestion
        ConcurrencyUtils.awaitLatch(startupLatch, "Timeout waiting for task to begin startup");
```

C0urante reviewed (commented):
Thanks @hgeraldino! Some final suggestions and then this will be good to go. I think it's important that these messages are accurate as they help illustrate the purpose of these tests and may save time for developers who have to read/rewrite them in the future.

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino committed:
Update connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

Co-authored-by: Chris Egerton <fearthecellos@gmail.com>

hgeraldino commented:
Committed the suggestions, thanks again for reviewing @C0urante!

BTW, I also opened https://github.com/apache/kafka/pull/13951 to have `WorkerSinkTaskThreadedTest` migrated. There's only one more in my bucket (WorkerSinkTaskTest) which has proven to be quite challenging

C0urante reviewed (commented):
Thanks @hgeraldino, LGTM!
